src/main/java/com/yelp/nrtsearch/server/luceneserver/doc/LoadedDocValues.java

```diff
@@ -494,4 +500,64 @@ public int size() {
       return values.size();
     }
   }
+
+  public static final class SingleVector extends LoadedDocValues<VectorType> {
+    private final BinaryDocValues docValues;
+    private VectorType value;
+
+    public SingleVector(BinaryDocValues docValues) {
+      this.docValues = docValues;
+    }
+
+    /**
+     * Set method to set the lucene level doc id to lookup value from index and initialize the
+     * loaded doc value index by loading vector data
+     */
+    public void setDocId(int docID) throws IOException {
+      if (docValues.advanceExact(docID)) {
+        value = decodeBytesRefToVectorType(docValues.binaryValue());
+      } else {
+        value = null;
+      }
+    }
+
+    /** Decodes binary doc value to float array and wraps it into a VectorType */
+    private static VectorType decodeBytesRefToVectorType(BytesRef bytesRef) {
+      byte[] byteArray = bytesRef.bytes;
+      float[] floats = new float[byteArray.length / Float.BYTES];
+      FloatBuffer fb = ByteBuffer.wrap(byteArray, 0, byteArray.length).asFloatBuffer();
+      fb.get(floats);
+      return new VectorType(floats);
+    }
+
+    /** Provide field value containing the doc value data for a given index */
+    @Override
+    public FieldValue toFieldValue(int index) {
+      VectorType vector = get(index);
+      Builder vectorBuilder =
+          Vector.newBuilder().addAllValue(Floats.asList(vector.getVectorData()));
```

Comment:
It might actually be better to do a for iteration over the vector data array and add the items individually. Otherwise it will end up boxing/unboxing the float values to Float Objects.